.github/workflows/ci.yaml

```diff
@@ -40,6 +40,47 @@ jobs:
       - run: poetry run invoke ${{ matrix.task }} --check
 
   tests:
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache-dependency-path: poetry.lock
+
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v4
+        with:
+          path: ~/.local # the path depends on the OS
+          key: poetry-${{ matrix.python-version  }} # increment to reset cache
+
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+
+      - name: Install dependencies
+        run: poetry install --sync
+
+      - name: Run tests
+        run: poetry run pytest -m "not agentjobs and not integration" --cov-report xml
+
+      # upload coverage report to codecov
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.1.0
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          slug: great-expectations/cloud
+          flags: ${{ matrix.python-version }},unit
```

Comment:
I'm not sure I did the codecov flags correctly. Originally I had separate flags e.g. `flags: ${{ matrix.python-version }},unit` which shows as:
<img width="1107" alt="image" src="https://github.com/great-expectations/cloud/assets/9903066/46a3f8ed-b38f-46f2-a8d2-64baa1c373cb">
Then changed to more of a namespace-like approach: `flags: unit-${{ matrix.python-version }}` and `flags: integration-${{ matrix.python-version }}`
Feedback welcome!

Reply:
I don't think we need to run these integration tests on every version of python.
The non-integration tests should catch any python version related problems.

I would just run the integration tests on a single version of python, and leave the python version out of the codecov flag. I would use whatever version we use for the container.